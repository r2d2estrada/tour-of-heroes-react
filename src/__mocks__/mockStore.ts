import configureStore from "redux-mock-store";
import { State } from "../redux";

const initialState: State = {
  heroes: [
    {
      id: 11,
      name: "Dr. Nice",
    },
    {
      id: 12,
      name: "Narco",
    },
    {
      id: 13,
      name: "Mr. Bombastic",
    },
  ],
  selectedHero: null,
  messages: [],
};

const middlewares: any = [];

const createMockStore = configureStore(middlewares);

export const mockStore = createMockStore(initialState);
