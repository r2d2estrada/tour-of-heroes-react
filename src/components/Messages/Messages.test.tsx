import React from "react";
import { render } from "@testing-library/react";
import Connected, { Messages } from ".";
import { Provider } from "react-redux";
import { mockStore } from "../../__mocks__/mockStore";

describe("Messages", () => {
  let cpnt: any;
  let connected: any;
  let mockData: any = ["Test 1", "Test 2"];

  beforeAll(() => {
    cpnt = <Messages messages={mockData} clearMessages={jest.fn()} />;
    connected = (
      <Provider store={mockStore}>
        <Connected />
      </Provider>
    );
  });

  it("should match snapshot", () => {
    const { container } = render(cpnt);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it("Should connect to redux", () => {
    const { container } = render(connected);
    expect(container).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { container } = render(cpnt);
  });
});
