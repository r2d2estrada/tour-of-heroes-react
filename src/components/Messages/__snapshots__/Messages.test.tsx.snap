// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-header"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h4>
            Messages:
          </h4>
        </div>
        <div
          class="col text-right"
        >
          <button
            class="btn btn-outline-secondary"
          >
            Clear
          </button>
        </div>
      </div>
    </div>
    <div
      class="card-body"
    >
      <ul
        class="list-group"
      >
        <li
          class="list-group-item"
        >
          Test 1
        </li>
        <li
          class="list-group-item"
        >
          Test 2
        </li>
      </ul>
    </div>
  </div>
</div>
`;
